gems/aws-sdk-core/spec/aws/plugins/retry_errors_spec.rb

```diff
@@ -6,97 +6,109 @@ module Plugins
+          expect(client.config.max_attempts).to eq(5)
+        end
 
-      it 'raises when max_attempts is not an integer' do
-        ENV['AWS_MAX_ATTEMPTS'] = 'string'
-        expect { client }.to raise_error(ArgumentError)
-      end
+        it 'defaults config.adaptive_retry_wait_to_fill to true' do
+          expect(client.config.adaptive_retry_wait_to_fill).to eq(true)
+        end
 
-      it 'raises when max_attempts is not >= 0' do
-        ENV['AWS_MAX_ATTEMPTS'] = -1
-        expect { client }.to raise_error(ArgumentError)
-      end
+        it 'can configure adaptive_retry_wait_to_fill using ENV with precedence over config' do
+          ENV['AWS_ADAPTIVE_RETRY_WAIT_TO_FILL'] = 'false'
+          expect(client.config.adaptive_retry_wait_to_fill).to eq(false)
+        end
 
-      it 'defaults config.adaptive_retry_wait_to_fill to true' do
-        expect(client.config.adaptive_retry_wait_to_fill).to eq(true)
-      end
+        it 'can configure adaptive_retry_wait_to_fill with shared config' do
+          allow_any_instance_of(Aws::SharedConfig)
+            .to receive(:adaptive_retry_wait_to_fill).and_return('false')
+          expect(client.config.adaptive_retry_wait_to_fill).to eq(false)
+        end
 
-      it 'can configure adaptive_retry_wait_to_fill using ENV with precedence over config' do
-        ENV['AWS_ADAPTIVE_RETRY_WAIT_TO_FILL'] = 'false'
-        expect(client.config.adaptive_retry_wait_to_fill).to eq(false)
-      end
+        it 'defaults config.correct_clock_skew to true' do
+          expect(client.config.correct_clock_skew).to eq(true)
+        end
 
-      it 'can configure adaptive_retry_wait_to_fill with shared config' do
-        allow_any_instance_of(Aws::SharedConfig)
-          .to receive(:adaptive_retry_wait_to_fill).and_return('false')
-        expect(client.config.adaptive_retry_wait_to_fill).to eq(false)
-      end
+        it 'can configure correct_clock_skew using ENV with precedence over config' do
+          ENV['AWS_CORRECT_CLOCK_SKEW'] = 'true'
+          expect(client.config.correct_clock_skew).to eq(true)
+        end
 
-      it 'defaults config.correct_clock_skew to true' do
-        expect(client.config.correct_clock_skew).to eq(true)
+        it 'can configure correct_clock_skew with shared config' do
+          allow_any_instance_of(Aws::SharedConfig)
+            .to receive(:correct_clock_skew).and_return('true')
+          expect(client.config.correct_clock_skew).to eq(true)
+        end
       end
 
-      it 'can configure correct_clock_skew using ENV with precedence over config' do
-        ENV['AWS_CORRECT_CLOCK_SKEW'] = 'true'
-        expect(client.config.correct_clock_skew).to eq(true)
-      end
+      context 'ENV is not stubbed' do
+        it 'can configure max_attempts using ENV with precedence over config' do
+          allow_any_instance_of(Aws::SharedConfig)
+            .to receive(:max_attempts).and_return(3)
+          ENV['AWS_MAX_ATTEMPTS'] = '1'
```

Comment:
I think it makes sense to have three tests here - tests for '1', '-1', and 1. We currently have 1 and -1, so we would need to make the change accordingly.